cocotb/share/makefiles/Makefile.inc

```diff
@@ -64,9 +64,13 @@ PYTHON_BIN ?= $(realpath $(shell cocotb-config --python-bin))
 include $(COCOTB_SHARE_DIR)/makefiles/Makefile.deprecations
 
 LIB_DIR=$(COCOTB_PY_DIR)/cocotb/libs
+LIB_PREFIX := lib
 
 ifeq ($(OS),Msys)
     LIB_EXT := dll
+    ifneq (,$(wildcard $(LIB_DIR)/cocotb.dll))
```

Comment:
I do not have a very strong opinion here but is a good idea to have the name of vpi/fli/vhpi different only for one simulator.

Reply:
That's not for a simulator here, mingw/msys uses the unix style `lib` prefix while microsoft convention on windows is without. At the moment this is mostly a change internal to cocotb but if `libgpi` or other components might be distributed separately, it may be a slight advantage to follow the native way.

Reply:
Will it complicate potential extensions?

Reply:
>That's not for a simulator here, mingw/msys uses the unix style lib prefix while microsoft convention on windows is without.

That is the point so now it is just for windows and msvc different. 

Is this really an advantage?

Reply:
Depending on the system that we use to load extensions, it might not be a problem. Since we are supporting full library names and paths in `GPI_EXTRA`, the file names in the list could be absolute paths to files on the system. These names would probably be populated with globing, so there is no issue of conditional computation of the library name.

I think our current use of `GPI_EXTRA` for loading "other" libraries for mixed language support will go the way of the dinosaur; and most of the library name computing added in this PR will be removed.

Reply:
Maybe can add this to `cocotb-config` so custom flows is a bit easier to build?
Something like `cocotb-config --vpi-lib icarus`?

Reply:
I think so. `cocotb-config` should be a one stop shop for custom flows. Maybe we can do that as a follow on.